app/services/medical_copays/vbs/service.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module MedicalCopays
+  module VBS
+    ##
+    # Service object for isolating dependencies in the {MedicalCopaysController}
+    #
+    # @!attribute request
+    #   @return [MedicalCopays::Request]
+    # @!attribute request_data
+    #   @return [RequestData]
+    # @!attribute response_data
+    #   @return [ResponseData]
+    class Service
+      attr_reader :request, :request_data, :response_data
+
+      ##
+      # Builds a Service instance
+      #
+      # @param opts [Hash]
+      # @return [Service] an instance of this class
+      #
+      def self.build(opts = {})
+        new(opts)
+      end
+
+      def initialize(opts)
+        @request = MedicalCopays::Request.build
+        @request_data = RequestData.build(user: opts[:user])
+        @response_data = ResponseData
```

Comment:
Do you need to call `ResponseData.build `here

Reply:
Moving the ResponseData and calling build on it directly inside the `get_copays` method. 👍